src/app/components/match/match.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { MatchService } from '@services/match.service';
+import { ResultService } from '@services/result.service';
+import { Match } from '@models/match';
+import { Team } from '@models/team';
+import { Set } from '@models/set'
+
+@Component({
+  selector: 'app-match',
+  templateUrl: './match.component.html',
+  styleUrls: ['./match.component.scss']
+})
+export class MatchComponent implements OnInit {
+
+  @Input() match: Match
+  @Input() teams: Team[]
+
+
+  constructor(
+    private matchService: MatchService,
+    private resultService: ResultService,
+  ) { }
+
+  ngOnInit() {
+  }
+
+  getTeamNameById(id: number, teams: Team[]): string {
+    return teams.find(team => team.id == id).name
+  }
+
+  showReferee(id: number, teams:Team[]): string {
+    return id ? this.getTeamNameById(id, teams) : "No referee set"
+  }
+
+  isMatchPlayed(sets: Set[]): boolean {
```

Comment:
I think it would make more sense to send in the whole match if we are asking if the match is played, rather than just the sets.

Reply:
Have changed this